report_qweb_pdf_watermark/models/report.py

```diff
@@ -27,7 +27,13 @@
 class Report(models.Model):
     _inherit = "ir.actions.report"
 
-    pdf_watermark = fields.Binary("Watermark")
+    use_company_watermark = fields.Boolean(
```

Comment:
A migration script is needed for this case

Reply:
@etobella Could you explain why an migration script would be needed?
The field pdf_watermark still exsists. See line 43
https://github.com/OCA/reporting-engine/compare/14.0...bosd:reporting-engine:15.0-mig-report_qweb_pdf_watermark#diff-66dceb063022cd58404dbbbb4649ffec677cbc89be10c385ad5714231929a220R34

The other field are a new addition, and should not cause a problem when migrating, right?

Reply:
If you migrate you will need to access all reports with watermarks in order to set the boolean to true. That is the migration needed...

Reply:
Oh, now I think I understand. 
By toggling the boolean you want to migrate the exsisting watermarks on the reports to the new `use_company_watermark` function.
I don’t think we want a migration script for that 😄 

The use_company_watermark is an additional new function. 🎉 
The current functionality of the pdf_watermark which can be defined on a per report basis is kept as is.

In the current situation after migrating to V15 the watermark on these reports stay unchanged.
So the result on V15 will look exactly the same as on V14. This is what we want. 👍 


For example if we would like to automatically migrate exsisting reports to the new use_company_watermark function things will get complicated and error prone.

Example:
If this module is installed on a V14 database one could have the situation that there is a different watermark specified for a SO report then a Warehouse out report.

E.g.
An SO report could have set an pdf_watermark.
`mywatermark1.pdf`

The Warehouse delivery report could have `mywatermark2.pdf`

Migrating existing pdf watermarks to the new use_global_watermark function is becoming complicated very quickly and could lead to unwanted results when done by a migration script.
In above situation the script must decide to either set mywatermark1.pdf or mywatermark2.pdf as the global company watermark.
If there are even more then 3 watermark pdf’sor custom regexes used things get unnecessary complicated.


In that scenario, it is much easier for an human to change to the global company watermark function trough the UI for those who want to use it.

Reply:
@etobella Do you agree with with above explanation of the way of werking?
Can we merge this one?

Reply:
ping @etobella

Reply:
Ouch, I see, sorry for the misunderstanding

Reply:
NP, Thanks for the review/merge!!